### ISSUE TITLE
AppRole Token Info

### ISSUE THREAD

**Issue by daborg69** [rajanadar/VaultSharp#23]
Hi, very nice library.  Thanks for all your work.  

I am working with AppRoles right now and trying to retrieve the current token's lease duration and lifetime so I can schedule a renew.  I have tried the following, but nothing returns any of the required information.

`           
 IAuthenticationInfo appRoleAuthenticationInfo = new AppRoleAuthenticationInfo(roleID, secretID);
IVaultClient vaultClient = VaultClientFactory.CreateVaultClient(x, appRoleAuthenticationInfo);

            var xx = vaultClient.GetCallingTokenInfoAsync();
            var yy = vaultClient.GetTokenRoleInfoAsync("abc-role");
            var zz = vaultClient.GetTokenRoleInfoAsync("def-role");
            
            var z1 = vaultClient.AppRoleAuthenticationGetRolesAsync();`

The variable xx has the following in it after the call.  
-		xx	Id = 21, Status = WaitingForActivation, Method = "{null}", Result = "{Not yet computed}"	System.Threading.Tasks.Task<VaultSharp.Backends.System.Models.Secret<VaultSharp.Backends.Authentication.Models.Token.CallingTokenInfo>>

It seems from another prior issue that this should be returning a data variable and a bunch of other information...  

Thanks for your assistance.

**Comment by rajanadar** [comment]
@daborg69 thanks for using the library.

the reason you see `xx` that way is because you are invoking an async method and not awaiting the results. So instead of getting the `Secret<TokenInfo>` you get a `Task<Secret<TokenInfo>>`.

In order to see the actual results, please `await` the call as follows:

        var xx = await vaultClient.GetCallingTokenInfoAsync();

This means that the method that holds your code also needs to be `async`.

If for some reason, you cannot do `async` methods, then there is a workaround as follows:

        var xx = vaultClient.GetCallingTokenInfoAsync().Result;

but please note that invoking `.Result` on a `Task` could result in deadlocks etc.

in summary, you need to either `await` the call OR use a synchronous invocation to get the results.
please see https://github.com/rajanadar/VaultSharp/blob/master/test/VaultSharp.UnitTests/AcceptanceTests.cs for more examples of invoking the library methods in an async manner.

let me know if any other issues.

**Comment by rajanadar** [comment]
some more info here:

https://github.com/rajanadar/VaultSharp#all-the-methods-are-async-how-do-i-use-them-synchronously

**Comment by daborg69** [comment]
Thanks that fixed it.

**Comment by rajanadar** [comment]
happy to hear that.
